crates/ruff_linter/src/rules/ruff/rules/snapshots/ruff_linter__rules__ruff__rules__unreachable__tests__for.py.md.snap

```diff
@@ -0,0 +1,405 @@
+### Source
+```python
+def func():
+    for i in range(13):
+        break
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1[["Loop continue"]]
+  block2["break\n"]
+  block3["for i in range(13):
+        break\n"]
+
+  start --> block3
+  block3 -- "range(13)" --> block2
+  block3 -- "else" --> block0
+  block2 --> return
+  block1 --> block3
+  block0 --> return
+```
+
+## Function 7
+### Source
+```python
+def func():
+    for i in range(1110):
+        if True:
+            break
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1[["Loop continue"]]
+  block2["break\n"]
+  block3["if True:
+            break\n"]
+  block4["for i in range(1110):
+        if True:
+            break\n"]
+
+  start --> block4
+  block4 -- "range(1110)" --> block3
+  block4 -- "else" --> block0
+  block3 -- "True" --> block2
+  block3 -- "else" --> block1
+  block2 --> return
+  block1 --> block4
+  block0 --> return
+```
+
+## Function 8
+### Source
+```python
+def func():
+    for i in range(5):
+        pass
+    else:
+        return 1
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1["return 1\n"]
+  block2[["Loop continue"]]
+  block3["pass\n"]
+  block4["for i in range(5):
+        pass
+    else:
+        return 1\n"]
+
+  start --> block4
+  block4 -- "range(5)" --> block3
+  block4 -- "else" --> block1
+  block3 --> block2
+  block2 --> block4
+  block1 --> return
+  block0 --> return
+```
+
+## Function 9
+### Source
+```python
+def func():
+    for i in range(5):
+        pass
+    else:
+        return 1
+    x = 1
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0["x = 1\n"]
+  block1["return 1\n"]
+  block2[["Loop continue"]]
+  block3["pass\n"]
+  block4["for i in range(5):
+        pass
+    else:
+        return 1\n"]
+
+  start --> block4
+  block4 -- "range(5)" --> block3
+  block4 -- "else" --> block1
+  block3 --> block2
+  block2 --> block4
+  block1 --> return
+  block0 --> return
+```
+
+## Function 10
+### Source
+```python
+def func():
+    for i in range(5):
+        pass
+    else:
+        pass
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1["pass\n"]
+  block2[["Loop continue"]]
+  block3["pass\n"]
+  block4["for i in range(5):
+        pass
+    else:
+        pass\n"]
+
+  start --> block4
+  block4 -- "range(5)" --> block3
+  block4 -- "else" --> block1
+  block3 --> block2
+  block2 --> block4
+  block1 --> block0
+  block0 --> return
+```
+
+## Function 11
+### Source
+```python
+def func():
+    for i in range(3):
+        if i == 2:
+            assert i is not None
+            break
+    else:
+        raise Exception()
+    x = 0
```

Comment:
Let's add an example with a nested loop where both the outer and inner loop use break/continue.